pkg/controller/nodemaintenance/check_lease_update.go

```diff
@@ -0,0 +1,312 @@
+package nodemaintenance
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+    . "github.com/onsi/ginkgo/extensions/table"
+	corev1 "k8s.io/api/core/v1"
+	coordv1beta1 "k8s.io/api/coordination/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	apitypes "k8s.io/apimachinery/pkg/types"
+	"time"
+	"context"
+)
+
+
+const NowTime = 100000
+
+
+func compareString(rval *string, lval *string) bool {
+	return (rval == nil && lval == nil) || (rval != nil && lval != nil && *rval == *lval)
+}
+
+func compareTime(rval *metav1.MicroTime, lval *metav1.MicroTime) bool {
+	return (rval == nil && lval == nil) || (rval != nil && lval != nil && (*rval).Time == (*lval).Time)
+}
+
+func compareInt32(rval *int32, lval *int32) bool {
+	return (rval == nil && lval == nil) || (rval != nil && lval != nil && *rval == *lval)
+}
+
+func makeString(val string) (*string) {
+	return &val
+}
+
+
+func getTimeLease(tmsec int64) (*metav1.MicroTime) {
+	tm  := time.Unix(int64(tmsec),0)
+	ret := metav1.MicroTime{Time: tm}
+	return &ret
+}
+
+func getMockNode() (*corev1.Node) {
+	node := &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "miau",
+					UID:  "foobar",
+				},
+	}
+	return node
+}
+
+var _ = Describe("checkLeaseUpdate", func() {
+
+	DescribeTable("check lease supported",
+		func(initLease *coordv1beta1.Lease, leasePostcondition *coordv1beta1.Lease) {
+			node := getMockNode()
+			objs := []runtime.Object{
+				initLease,
+			}
+			cl  := fake.NewFakeClient(objs...)
+			timeNow :=  time.Unix(NowTime,0)
```

Comment:
Seems like NowTime would be a good candidate to pass in as an argument to func()

Reply:
poke?